### ISSUE TITLE
BlackMagicProbe and t28 NRF52 smartwatch

### ISSUE THREAD

**Issue by batuakan** [micooke/arduino-nRF5-smartwatches#10]
Hi Mark,  

Recently I acquired an T28 smartwatch and i was hoping to upload a custom firmware to repurpose the device.

I followed you steps, and soldered wires to the GND, SWDIO and SWCLK pinouts as mentioned in your photographs of the T28, and left the device battery connected and tried to connect to the device with Black Magic Probe (BMP). Using arm-none-eabi-gdb,
(gdb) target extended-remote COM5
Remote debugging using COM5
(gdb) mon swdp_scan
Target voltage: unknown
SW-DP scan failed!

I reversed swdio and swclk but still no success :(

I connect my bmp to an older nrf51 board and it works perfectly fine.

Does the device need to be in a special state to accept this connection? is this a BMP issue or is it a T28 issue?

**Comment by micooke** [comment]
Hi @batuakan! Sounds like a connection issue. I wired mine up a LONG time ago, so I might have missed something, but that should be everything you need. It may be something silly like you need to give it +5v as well.
My t28 has had an autopsy, and i know that i put a switch inline with the battery but i don't remember why (maybe so I could reset it easily?) I'll have a play in the next day or so and get back to you with photos. Cheers!

**Comment by micooke** [comment]
I checked my configuration and my BMP finds it with and without +5v connected so i dont know what to suggest.
One other thing to note is that you will need to disable the read protection, which i only got working with a J Link : https://github.com/micooke/arduino-nRF5-smartwatches/blob/master/nrf52_disable_read_protection.txt
You can find J Links on aliexpress et al.

**Comment by batuakan** [comment]
A few updates:
I found out that my connection issues were due to using an older version of BMP, as the support for the NRF52 was added later. Now i use a never version and swdp_scan success returns a list of items.

I have mass erased the device, and my BMP successfully seemingly uploads firmware to the device, but i am not sure if it really succeeds, same happens with STLinkv2 . I will try to make BMP work with NRF52 before ordering the JLink. I will post here if i am successful
